trillian/ctfe/config.go

```diff
@@ -78,6 +79,63 @@ func MultiLogConfigFromFile(filename string) (*configpb.LogMultiConfig, error) {
 	return &cfg, nil
 }
 
+// ValidateLogConfig checks that a signle log config is valid. In particular:
+//  - A mirror log has a valid public key and no private key.
+//  - A non-mirror log has a private, and optionally a public key (both valid).
+//  - Each of NotBeforeStart and NotBeforeLimit, if set, is a valid timestamp
+//    proto. If both are set then NotBeforeStart <= NotBeforeLimit.
+//
+// TODO(pavelkalinnikov): Return the parsed values so that we don't have to
+// parse them again after validation.
+func ValidateLogConfig(cfg *configpb.LogConfig) error {
+	if cfg.LogId == 0 {
+		return errors.New("empty log ID")
+	}
+	var err error
+
+	// Validate the public key.
+	pubKey := cfg.PublicKey
+	if pubKey != nil {
+		if _, err = der.UnmarshalPublicKey(pubKey.Der); err != nil {
+			return fmt.Errorf("invalid public key: %v", err)
+		}
+	} else if cfg.IsMirror {
+		return errors.New("empty public key")
+	}
+
+	// Validate the private key.
+	if !cfg.IsMirror {
```

Comment:
Maybe it would be clearer to move the key validation to mirror / non mirror helpers, might cut down the if {} nesting.

Reply:
I tried that, didn't like it much. It introduced some duplication (e.g. public keys need to be validated in both cases), and the code became bulky. I'd rather keep the current code, as it's reasonably small.